stable/datadog/templates/daemonset-volumes-linux.yaml

```diff
@@ -7,9 +7,16 @@
   name: cgroups
 {{- if .Values.datadog.dogstatsd.useSocketVolume }}
 - hostPath:
-    path: "/var/run/datadog/"
+    path: {{ .Values.datadog.dogstatsd.hostSocketPath }}
+    type: Socket
   name: dsdsocket
 {{- end }}
+{{- if .Values.datadog.apm.useSocketVolume }}
+- hostPath:
+    path: {{ .Values.datadog.apm.hostSocketPath }}
+    type: Socket
+  name: apmsocket
```

Comment:
The risk in mounting directly the socket is to have the same issue that we had with the docker socket:
If the trace-agent is crashing and is restarting, it will rebind a new UNIX socket.
I fear that the applications that were connected to the previous instance of the trace-agent will keep on being connected to the orphan UNIX socket.